service/dap/server.go

```diff
@@ -1216,12 +1240,21 @@ func (s *Server) onSetBreakpointsRequest(request *dap.SetBreakpointsRequest) {
 		} else {
 			// Create new breakpoints.
 			got, err = s.debugger.CreateBreakpoint(
-				&api.Breakpoint{File: serverPath, Line: want.Line, Cond: want.Condition, HitCond: want.HitCondition, Name: reqString})
+				&api.Breakpoint{
+					File:       serverPath,
+					Line:       want.Line,
+					Cond:       want.Condition,
+					HitCond:    want.HitCondition,
+					Name:       reqString,
+					Tracepoint: want.LogMessage != "",
+					UserData:   want.LogMessage,
```

Comment:
For any breakpoint, logpoint or not, there is a message sent back to a client. In dap terms it could be an event and/or a response, so the nature of the breakpoint doesn't really matter. The low-level debugger already can run and hit breakpoints while no client is connected. That makes whatever handler was blocked while running exit and try to send messages back to a disconnected client. I think I have seen those client send errors in the rpc sessions. Dap send() doesn't log the send errors, just the read loop errors. It just silently fails. We could try to guard against these useless sends, but it doesn't crash the server, so likely not worth it. But a test case would be in order to make sure things stay that way. 

Also as per protocol, a dap client clears all existing breakpoints when connecting to a server.